java-security/README.md

```diff
@@ -141,50 +201,58 @@ if(result.isErroneous()) {
-java.security.Principal principal = token.getPrincipal();
-Instant expiredAt = token.getExpiration();
-String keyId = token.getHeaderParameterAsString(TokenHeader.KEY_ID);
-...
-```
+To use validation listeners with `XsuaaTokenAuthenticator` or `IasTokenAuthenticator`, follow the example below:
 
-### Get further information from `VCAP_SERVICES`
-In case you need further details from `VCAP_SERVICES` system environment variable, which are not exposed by `OAuth2ServiceConfiguration` interface you can use the `DefaultJsonObject` class for Json parsing. 
-
-Example:
 ```java
-String vcapServices = System.getenv(CFConstants.VCAP_SERVICES);
-JsonObject serviceJsonObject = new DefaultJsonObject(vcapServices).getJsonObjects(Service.XSUAA.getCFName()).get(0);
-Map<String, String> xsuaaConfigMap = serviceJsonObject.getKeyValueMap();
-Map<String, String> credentialsMap = serviceJsonObject.getJsonObject(CFConstants.CREDENTIALS).getKeyValueMap();
+XsuaaTokenAuthenticator authenticator = new XsuaaTokenAuthenticator().withValidationListener(ValidationListener);
 ```
 
-## Token based authentication
-The servlet authenticator part of this library makes it easy to integrate token based authentication into your java application.
-For the integration of different Identity Services the [`TokenAuthenticator`](/java-api/src/main/java/com/sap/cloud/security/servlet/TokenAuthenticator.java) interface was created. Right now there are these implementations:
-
-- [XsuaaTokenAuthenticator](src/main/java/com/sap/cloud/security/servlet/XsuaaTokenAuthenticator.java)
-- [IasTokenAuthenticator](src/main/java/com/sap/cloud/security/servlet/IasTokenAuthenticator.java)
-
-> Depending on the application's needs the `TokenAuthenticator` can be customized.
-
-![](images/xsuaaFilter.png)
+Alternatively, configure validation listeners using `JwtValidatorBuilder`:
 
+```java
+CombiningValidator<Token> validators = JwtValidatorBuilder
+		.getInstance(serviceConfig)
+		.withValidatorListener(ValidationListener) // add your Validation Listener
+		.build();
+```
 
-### ProofOfPossession validation 
+### ProofOfPossession validation
 #### X509 certificate thumbprint `X5t` validation
-[JwtX5tValidator](src/main/java/com/sap/cloud/security/token/validation/validators/JwtX5tValidator.java) offers JWT Certificate Thumbprint `X5t` confirmation method's validation. See specification [here](https://tools.ietf.org/html/rfc8705#section-3.1). 
-This validator is not part of the default `CombiningValidator`, it needs to be added manually to `JwtValidatorBuilder` to use it. 
+[JwtX5tValidator](src/main/java/com/sap/cloud/security/token/validation/validators/JwtX5tValidator.java) offers JWT Certificate Thumbprint `X5t` confirmation method's validation. See specification [here](https://tools.ietf.org/html/rfc8705#section-3.1).
+This validator is not part of the default `CombiningValidator`, it needs to be added manually to `JwtValidatorBuilder` to use it.
```

Comment:
Yes exactly it's only for specific use cases. `cnf` token claim is only provided by the IAS in case of certificate based authentication. As every application binding gets its own certificate the cnf thumbprint can only be validated by the application that was directly called with the certificate of the calling application.
Consequently, this does not work for e.g. forwarded tokens